subprojects/core/src/integTest/groovy/org/gradle/configuration/project/LifecycleProjectEvaluatorIntegrationTest.groovy

```diff
@@ -95,4 +98,39 @@ class LifecycleProjectEvaluatorIntegrationTest extends AbstractIntegrationSpec {
         failure.assertHasDescription("A problem occurred configuring root project 'root'.")
         failure.assertHasCause("after evaluate failure")
     }
+
+    def "captures lifecycle operations"() {
+        given:
+        settingsFile << """
+            include 'foo'
+        """
+        file("foo/build.gradle")
+        file("foo/before.gradle") << ""
+        file("foo/after.gradle") << ""
+        buildFile << """
+            project(':foo').beforeEvaluate {
+                project(':foo').apply from: 'before.gradle'
+            }
+            project(':foo').afterEvaluate {
+                project(':foo').apply from: 'after.gradle'
+            }
+        """
```

Comment:
a common pattern we see is to have a afterEvaluate block within a plugin application. We should mimic this here and verify the expected tree structure